syntax = "proto3";

package masterworker;

service WorkerService {
    rpc assignTask (WorkerQuery) returns (WorkerResponse){}
}

message WorkerQuery{
    string type = 1;
    string userid = 2;
    int32 n = 3; 
    string output = 4;
    repeated WorkerShard shards = 5;
    int32 workerid = 6;
}

message WorkerShard{
    string path = 1;
    int32 start = 2;
    int32 end = 3;
}

message WorkerResponse{
    int32 id = 1;
    int32 status = 2;
}